src/gg.regression.unity.experimental.runespawn/Editor/Scripts/PlayModeController.cs

```diff
@@ -10,6 +11,50 @@ public static class PlayModeController
 {
     private static bool originalRunInBackground = Application.runInBackground;
 
+    private static List<string> _logs = new List<string>();
+    private static List<string> _errors = new List<string>();
+
+    /// <summary>
+    /// Static constructor for PlayModeController.
+    /// Initializes event subscriptions and prevents multiple subscriptions.
+    /// </summary>
+    static PlayModeController()
+    {
+        // Prevent multiple subscriptions.
+        EditorApplication.playModeStateChanged -= OnPlayModeStateChanged;
```

Comment:
I would test this thoroughly..  If you ever get a case where you try to remove a listener that doesn't exist it will throw a Unity exception.

Reply:
I tested it in this context and it works. Let me if I might be missing something

```csharp
 public class BotTester : MonoBehaviour
    {
       UnityEvent m_MyEvent;
       void Start()
       {
            Debug.Log("BotTester: Started.");

            Application.logMessageReceived -= SomeRandomFunction;
            Debug.Log("BotTester: Removed subscription.");

            m_MyEvent = new UnityEvent();
            m_MyEvent.RemoveListener(SomeOtherRandomFunction);
            Debug.Log("BotTester: Removed listener.");
       }

        private void SomeRandomFunction(string condition, string stackTrace, LogType type){}
        private void SomeOtherRandomFunction(){}

    }
```